src/main/java/com/derongan/minecraft/mineinabyss/Relic/Behaviour/Behaviours/CampfireRelicBehaviour.java

```diff
@@ -0,0 +1,57 @@
+package com.derongan.minecraft.mineinabyss.Relic.Behaviour.Behaviours;
+
+import com.derongan.minecraft.mineinabyss.Relic.Behaviour.UseRelicBehaviour;
+import com.derongan.minecraft.mineinabyss.Relic.Relics.RelicType;
+import com.derongan.minecraft.mineinabyss.Relic.Relics.SlightlyOffzRelicType;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.*;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.util.EulerAngle;
+
+public class CampfireRelicBehaviour implements UseRelicBehaviour {
+    @Override
+    public void onUse(PlayerInteractEvent event) {
+        if (event.getAction().equals(Action.RIGHT_CLICK_BLOCK) && event.getBlockFace() == BlockFace.UP) {
+            event.setCancelled(true);
+
+            Player player = event.getPlayer();
+
+            Block target = event.getClickedBlock();
+
+            Material ma = target.getType();
+
+            if(ma.equals(Material.GRASS) || ma.equals(Material.STONE)) {
+                event.getPlayer().getInventory().getItemInMainHand().setAmount(event.getPlayer().getInventory().getItemInMainHand().getAmount() - 1);
+                doPlaceCampfire(target);
+            }
+        }
+    }
+
+    private void doPlaceCampfire(Block target) {
+        RelicType type = SlightlyOffzRelicType.CAMPFIRE;
```

Comment:
Make this more generic. All behaviours have the getRelicType method. Use this to get the relic this behaviour belongs to. This way I can create an "UglyCampfireRelicType" that also has this behaviour